src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3832,15 +3832,17 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
        * an error message is reported and None is returned.
        */
       def tree2ConstArg(tree: Tree, pt: Type): Option[ClassfileAnnotArg] = tree match {
-        case Apply(Select(New(tpt), nme.CONSTRUCTOR), args) if (pt.typeSymbol == ArrayClass) =>
+        case Apply(Select(New(tpt), nme.CONSTRUCTOR), args) if pt.typeSymbol == ArrayClass && !unit.isJava =>
```

Comment:
What do you think about adding this case instead?

```scala
        case Apply(Select(New(_), nme.CONSTRUCTOR), _) if pt.typeSymbol == ArrayClass && unit.isJava =>
          // In Java, a single value may be passed for array annotation parameters
          tree2ConstArg(Apply(Select(gen.mkAttributedRef(ArrayModule), nme.apply), List(tree)), pt)
```

Reply:
You've seen the case below for this, right?  "If we get here"

That idea sounds good if it subsumes the other one.  I'll try, thanks!

Reply:
ah no, i hadn't seen that case below

Reply:
either way works for me, shave off the array from the expected type, or wrap the expreesion in `Array(...)`

Reply:
Ah, no, they work together...